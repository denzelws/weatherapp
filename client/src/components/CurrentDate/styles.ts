import styled, { css } from 'styled-components'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    flex-direction: column;
    background: rgb(0, 151, 167);
    padding: ${theme.spacings.xxsmall} 0;
  `}
`

export const DateInformation = styled.div``

export const Date = styled.h3`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-weight: ${theme.font.light};
  `}
`
export const Place = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    margin-top: 1.2rem;
    display: flex;
    align-items: center;
    justify-items: center;
  `}
`

export const City = styled.h3`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-weight: ${theme.font.bold};
  `}
`
