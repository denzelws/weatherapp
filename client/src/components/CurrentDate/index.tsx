import * as S from './styles'

const CurrentDate = () => (
  <S.Wrapper>
    <S.Date>Quinta, 27 Outubro 2022 | Hor√°rio Local: 23:00 PM</S.Date>
    <S.Place>
      <S.City> Rio, BR</S.City>
    </S.Place>
  </S.Wrapper>
)

export default CurrentDate
