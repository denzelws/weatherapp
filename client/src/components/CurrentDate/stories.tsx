import { Story, Meta } from '@storybook/react'
import CurrentDate from '.'

export default {
  title: 'CurrentDate',
  component: CurrentDate
} as Meta

export const Default: Story = () => <CurrentDate />
