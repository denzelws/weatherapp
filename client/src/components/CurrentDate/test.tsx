import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import CurrentDate from '.'

describe('<CurrentDate />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<CurrentDate />)
    expect(screen.getByRole('heading', { name: /quinta/i })).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /rio, br/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
