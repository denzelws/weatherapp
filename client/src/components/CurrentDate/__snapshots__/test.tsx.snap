// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentDate /> should render the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: rgb(0,151,167);
  padding: 2.4rem 0;
}

.c1 {
  color: #FAFAFA;
  font-weight: 300;
}

.c2 {
  color: #FAFAFA;
  margin-top: 1.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: center;
}

.c3 {
  color: #FAFAFA;
  font-weight: 600;
}

<main
  class="c0"
>
  <h3
    class="c1"
  >
    Quinta, 27 Outubro 2022 | Hor√°rio Local: 23:00 PM
  </h3>
  <div
    class="c2"
  >
    <h3
      class="c3"
    >
       Rio, BR
    </h3>
  </div>
</main>
`;
